src/api/getSwap.ts

```diff
@@ -0,0 +1,103 @@
+import axios from "axios";
+import apiEndpoint from "./apiEndpoint";
+
+export interface Asset {
```

Comment:
:mega: Please use the types we already have.

Reply:
Do you mean the ones in `getSwaps.ts`? I will combine them!

Reply:
I think there are more generic ones somewhere in the codebase. You should rely on `quantity` for example. Not every asset is guaranteed to have that field :)

Reply:
> I think there are more generic ones somewhere in the codebase.

I think you're referring to what we do in [ledgerSpec.ts](https://github.com/comit-network/comit-i/blob/8ab8c4dff0eb38f85ec7659c50a3e6e5922663a0/src/ledgerSpec.ts), but I'm not sure how to apply that here. I don't know how to handle different shapes for the asset other than using optional fields. For now, I would just add `token_contract` as an optional field and use it if it's present and the asset name is `erc20`.